### ISSUE TITLE
Array of composite type inside composite type

### ISSUE THREAD

**Issue by alaneos777** [pgcentralfoundation/pgrx#1293]
Hi everyone, I was trying to follow this [example](https://github.com/pgcentralfoundation/pgrx/blob/master/pgrx-examples/composite_type/src/lib.rs) and create a composite type that holds an array of another composite type. This is the whole code:

```rust
use pgrx::prelude::*;

pgrx::pg_module_magic!();

extension_sql!(
    r#"
    CREATE TYPE entity AS (id INT, lang TEXT);
    CREATE TYPE result AS (score DOUBLE PRECISION, entities entity[]);
    "#,
    name = "init_example",
    bootstrap
);

#[pg_extern]
fn create_result() -> pgrx::composite_type!("result") {
    let mut record = PgHeapTuple::new_composite_type("result").unwrap();
    record.set_by_name("score", 0.707).unwrap();
    let mut entity_records: Vec<pgrx::composite_type!("entity")> = Vec::new();
    for i in 0..10 {
        let mut entity_record = PgHeapTuple::new_composite_type("entity").unwrap();
        entity_record.set_by_name("id", i).unwrap();
        entity_record.set_by_name("lang", "en".to_string()).unwrap();
        entity_records.push(entity_record);
    }
    record.set_by_name("entities", entity_records).unwrap();
    return record;
}

/// This module is required by `cargo pgrx test` invocations.
/// It must be visible at the root of your extension crate.
#[cfg(test)]
pub mod pg_test {
    pub fn setup(_options: Vec<&str>) {
        // perform one-off initialization when the pg_test framework starts
    }

    pub fn postgresql_conf_options() -> Vec<&'static str> {
        // return any postgresql.conf settings that are required for your tests
        vec![]
    }
}
```

But when I want to call `select * from create_result();`, I got this:
```
example=# create extension example;
CREATE EXTENSION
example=# select * from create_result();
ERROR:  called `Result::unwrap()` on an `Err` value: IncompatibleTypes { rust_type: "alloc::vec::Vec<pgrx::heap_tuple::PgHeapTuple<pgrx::pgbox::AllocatedByRust>>", rust_oid: Oid(25463), datum_type: "record[]", datum_oid: Oid(2287) }
example=#
```

What do I need to change to achieve this scenario? Thanks in advance for your help.

**Comment by eeeebbbbrrrr** [comment]
Thanks for the report.  We'll take a look at the bug.

**Comment by eeeebbbbrrrr** [comment]
This was fixed in PR #1303 and we'll get a new point release out soon.

**Comment by workingjubilee** [comment]
Released in v0.10.2!
